language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FailStatementNodeContext.java

```diff
@@ -57,4 +67,67 @@ public FailStatementNodeContext() {
         
         return completionItems;
     }
+
+    @Override
+    public void sort(BallerinaCompletionContext context, FailStatementNode node,
+                     List<LSCompletionItem> completionItems) {
+        for (LSCompletionItem lsCItem : completionItems) {
+            if (lsCItem.getType() == LSCompletionItem.CompletionItemType.SYMBOL) {
+                SymbolCompletionItem symbolCompletionItem = (SymbolCompletionItem) lsCItem;
+                if (isCompletionItemSubTypeOfError(symbolCompletionItem)) {
+                    lsCItem.getCompletionItem().setSortText(SortingUtil.genSortText(1));
+                } else {
+                    lsCItem.getCompletionItem().setSortText(SortingUtil.genSortText(3));
+                }
+            } else if (SortingUtil.isModuleCompletionItem(lsCItem)) {
+                lsCItem.getCompletionItem().setSortText(SortingUtil.genSortText(2));
+            } else {
+                lsCItem.getCompletionItem().setSortText(SortingUtil.genSortText(3));
+            }
+        }
+    }
+
+    private boolean isUnionOfErrors(UnionTypeSymbol unionTypeSymbol) {
+        List<TypeSymbol> typeSymbols = unionTypeSymbol.memberTypeDescriptors();
+        return typeSymbols.stream()
+                .allMatch(typeSymbol -> CommonUtil.getRawType(typeSymbol).typeKind() == TypeDescKind.ERROR);
+    }
+
+    private boolean isFunctionReturnsUnionOfErrors(FunctionTypeSymbol functionTypeSymbol) {
+        Optional<TypeSymbol> retTypeDec = functionTypeSymbol.returnTypeDescriptor();
+        if (retTypeDec.isEmpty()) {
+            return false;
+        }
+        TypeSymbol rawType = CommonUtil.getRawType(retTypeDec.get());
+        if (rawType.typeKind() == TypeDescKind.UNION) {
+            UnionTypeSymbol unionTypeSymbol = (UnionTypeSymbol) rawType;
+            return isUnionOfErrors(unionTypeSymbol);
+        }
+        return rawType.typeKind() == TypeDescKind.ERROR;
+    }
+    
+    private boolean isCompletionItemSubTypeOfError(SymbolCompletionItem symbolCompletionItem) {
+        Optional<Symbol> symbol = symbolCompletionItem.getSymbol();
+        if (symbol.isEmpty()) {
+            return false;
+        }
+        Optional<TypeSymbol> tSymbol = SymbolUtil.getTypeDescriptor(symbol.get());
+        if (symbol.get().kind() == SymbolKind.TYPE_DEFINITION) {
+            return false;
+        }
+
+        if (tSymbol.isPresent()) {
+            TypeSymbol typeSymbol = CommonUtil.getRawType(tSymbol.get());
+            if (typeSymbol.typeKind() == TypeDescKind.ERROR) {
```

Comment:
After having an early return, change the order of the if-clauses. Get union and function check first and at the else clause we can use `return typeSymbol.typeKind() == TypeDescKind.ERROR`. WDYT?

Reply:
Agree +1